### PR TITLE
Devnet

### PR DESCRIPTION
**Pull request by vedhavyas**
This PR brings following changes:
- Devnet
- Ability to prune nodes
- Farmer node scripts to run farmer

~Note: Still need to update docker tags but good to review~

### CODE REVIEW

File: network-primitives/scripts/create_farmer_node_compose_file.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+  farmer:
+    depends_on:
+      archival-node:
+        condition: service_healthy
+    image: ghcr.io/\${NODE_ORG}/farmer:\${NODE_TAG}
+    volumes:
+      - farmer_data:/var/subspace:rw
+    restart: unless-stopped
+    command: [
```
Comment (nazar-pc) [inline network-primitives/scripts/create_farmer_node_compose_file.sh]: Farmer should also expose DSN port or you didn't want that?

File: .gitignore [hunk 2]
```diff
@@ -18,3 +18,9 @@ node_keys.txt
 /resources/gemini-3c/full_node_keys.txt
 /resources/gemini-3c/rpc_node_keys.txt
 /resources/gemini-3c/relayer_ids.txt
+/resources/devnet/relayer_ids.txt
```
Comment (nazar-pc) [inline .gitignore:21]: What is relayer doing here?

**Review by nazar-pc — DISMISSED** [review]

File: .gitignore (hunk 2 above)
Comment (vedhavyas) [reply .gitignore:21]: this hold the relayer ids used to run the relayers for domains

File: network-primitives/scripts/create_farmer_node_compose_file.sh (hunk 1 above)
Comment (vedhavyas) [reply network-primitives/scripts/create_farmer_node_compose_file.sh]: I was not aware farmer can. Thought not sure if we want to. Lets leave it as is for now

Comment (nazar-pc) [reply network-primitives/scripts/create_farmer_node_compose_file.sh]: We have an example with ports in the farming docs with Docker in monorepo

**Review by nazar-pc — APPROVED** [review]

**Pull request merged**